prometheus_hardware_exporter/collector.py

```diff
@@ -109,3 +110,386 @@ def fetch(self) -> List[Payload]:
     def process(self, payloads: List[Payload], datastore: Dict[str, Payload]) -> List[Payload]:
         """Process the payload if needed."""
         return payloads
+
+
+class LSISAS2ControllerCollector(BlockingCollector):
+    """Collector for LSI SAS-2 controllers."""
+
+    sas2ircu = Sasircu(2)
+
+    @property
+    def specifications(self) -> List[Specification]:
+        """Define LSI SAS-2 metric specs."""
+        return [
+            Specification(
+                name="lsi_sas_2_controllers",
+                documentation="Number of LSI SAS-2 controllers",
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_ir_volumes",
+                documentation="Number of IR volumes",
+                labels=["controller_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_ready_ir_volumes",
+                documentation="Number of ready IR volumes",
+                labels=["controller_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_unready_ir_volumes",
+                documentation="Number of unready IR volumes",
+                labels=["controller_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_ir_volume",  # will append "_info" internally
+                documentation="Shows the information about the integrated RAID volume",
+                metric_class=InfoMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_ready_physical_devices",
+                documentation="Number of ready physical devices",
+                labels=["controller_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_unready_physical_devices",
+                documentation="Number of unready physical devices",
+                labels=["controller_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
```

Comment:
According to https://docs.google.com/spreadsheets/d/1KiQ5cS36zfjf8V32RoD1RKx9q01AhgoKvBQRkonQv5o/edit#gid=1149993398
There is a metrics named "lsi_sas_2_physical_devices", but it is not available here.

Reply:
Nice catch! I missed it, Thanks.